### PR TITLE
Ten more clients added 

### PR DESCRIPTION
**Pull request by luchianenco**
Let's increase the number to 45 ;)
Box, Buffer, CanvasLMS, Clever, DevianArt, Discord, Elance, Eventbrite, Foursquare and Mail.Ru are added

I use this list of providers [http://oauth2-client.thephpleague.com/providers/thirdparty/](url).
Maybe you have another one? Thx

### CODE REVIEW

File: src/DependencyInjection/Providers/CanvasLMSProviderConfigurator.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            ->scalarNode('canvas_instance_url')
+                ->isRequired()
+                ->info('URL of Canvas Instance (e.g. https://canvas.instructure.com)')
+            ->end()
+            ->scalarNode('purpose')
+                ->defaultValue('')
+                ->info(
+                    'This can be used to help the user identify which instance of an application this token is '
+                    .'for. For example, a mobile device application could provide the name of the device.'
+                )
```
Comment (weaverryan) [inline src/DependencyInjection/Providers/CanvasLMSProviderConfigurator.php]: Nice!

File: src/DependencyInjection/Providers/CanvasLMSProviderConfigurator.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        return 'smtech\OAuth2\Client\Provider\CanvasLMS';
+    }
+
+    public function getProviderOptions(array $config)
+    {
+        return [
+            'clientId' => $config['client_id'],
+            'clientSecret' => $config['client_secret'],
+            'canvasInstanceUrl' => $config['canvas_instance_url']
+        ];
```
Comment (weaverryan) [inline src/DependencyInjection/Providers/CanvasLMSProviderConfigurator.php]: I think the `purpose` needs to be mentioned here?

Comment (luchianenco) [reply src/DependencyInjection/Providers/CanvasLMSProviderConfigurator.php]: oh, forgot :(

**Comment by luchianenco** [comment]
You are right!!! Doc generator is really nice tool ... thanks for it ;)

**Comment by bocharsky-bw** [comment]
Oh my... where do you find them? I even have not heard about some of them :p Great job!

> You're making me so happy that we made the documentation auto-generate

Lol, that's true :)

**Comment by luchianenco** [comment]
@bocharsky-bw your are right .... some of them are really unknown ... and I think sometimes : do we really need them all or not :)

**Comment by bocharsky-bw** [comment]
Good question! IMO having them all is fine until their libraries are supported / not deprecated. I'd say it worth us nothing to have them all... well, with your BIG help, it worth us nothing ;)

**Comment by luchianenco** [comment]
Good, I will add them all from time to time, and then I will try to update deprecated libraries that are a bit popular :)

**Comment by weaverryan** [comment]
Sweeeeet!

**Pull request merged**